src/lib/ngsi10/UpdateContextRequest.cpp

```diff
@@ -288,6 +288,10 @@ void UpdateContextRequest::fill
   else
   {
     caP = new ContextAttribute(attributeName, ucarP->type, ucarP->contextValue);
+    if (ucarP->valueType == orion::ValueTypeNone)
+    {
+      caP->valueType = orion::ValueTypeNone;
+    }
   }
```

Comment:
From a logical point of view, it makes sense... as this fragment is covering what the construct is not able to do, so it should be an unconditional copy.

I'd modify and re-run test. If works, I would use your suggested approach.